Jenkinsfile

```diff
@@ -1,103 +1,140 @@
+        steps {
+          echo "Testing ${env.UBUNTU_GNU_IMAGE_TAG}"
+          sh 'docker run "${UBUNTU_GNU_IMAGE_TAG}" gfortran --version'
+        }
       }
-    }
-
-    stage('Build Ubuntu GNU HPC Stack') {
-      steps {
-        echo "Building ${env.UBUNTU_GNU_HPCSTACK_IMAGE_TAG}"
-        sh 'docker build --tag "${UBUNTU_GNU_HPCSTACK_IMAGE_TAG}" --file "${WORKSPACE}/Docker/Dockerfile.${UBUNTU_GNU_HPCSTACK_IMAGE_NAME}" "${WORKSPACE}"'
+  
+      stage('Release Ubuntu GNU') {
+        when {
+          branch 'main'
+        }
+  
+        steps {
+          echo "Releasing ${env.UBUNTU_GNU_IMAGE_TAG}"
+          sh 'docker push "${UBUNTU_GNU_IMAGE_TAG}"'
+        }
+      }
+  
+      stage('Build Ubuntu GNU HPC Stack') {
+        steps {
+          echo "Building ${env.UBUNTU_GNU_HPCSTACK_IMAGE_TAG}"
+          sh 'docker build --tag "${UBUNTU_GNU_HPCSTACK_IMAGE_TAG}" --file "${WORKSPACE}/Docker/Dockerfile.${UBUNTU_GNU_HPCSTACK_IMAGE_NAME}" "${WORKSPACE}"'
+        }
+      }
+  
+      stage('Test Ubuntu GNU HPC Stack') {
+        steps {
+          echo "Testing ${env.UBUNTU_GNU_HPCSTACK_IMAGE_TAG}"
+          sh 'docker run "${UBUNTU_GNU_HPCSTACK_IMAGE_TAG}"'
+        }
+      }
+  
+      stage('Release Ubuntu GNU HPC Stack') {
+        when {
+          branch 'main'
+        }
+  
+        steps {
+          echo "Releasing ${env.UBUNTU_GNU_HPCSTACK_IMAGE_TAG}"
+          sh 'docker push "${UBUNTU_GNU_HPCSTACK_IMAGE_TAG}"'
+        }
+      }
+  
+      stage('Build Ubuntu GNU SRW') {
+        steps {
+          echo "Building ${env.UBUNTU_GNU_SRW_IMAGE_TAG}"
+          sh 'docker build --tag "${UBUNTU_GNU_SRW_IMAGE_TAG}" --file "${WORKSPACE}/Docker/Dockerfile.${UBUNTU_GNU_SRW_IMAGE_NAME}" "${WORKSPACE}"'
       }
     }
 
-    stage('Test Ubuntu GNU HPC Stack') {
+    stage('Test Ubuntu GNU SRW') {
       steps {
-        echo "Testing ${env.UBUNTU_GNU_HPCSTACK_IMAGE_TAG}"
-        sh 'docker run "${UBUNTU_GNU_HPCSTACK_IMAGE_TAG}"'
+        echo "Testing ${env.UBUNTU_GNU_SRW_IMAGE_TAG}"
+        sh 'docker run "${UBUNTU_GNU_SRW_IMAGE_TAG}"'
       }
     }
 
-    stage('Release Ubuntu GNU HPC Stack') {
+    stage('Release Ubuntu GNU SRW') {
       when {
         branch 'main'
       }
 
       steps {
-        echo "Releasing ${env.UBUNTU_GNU_HPCSTACK_IMAGE_TAG}"
-        sh 'docker push "${UBUNTU_GNU_HPCSTACK_IMAGE_TAG}"'
+        echo "Releasing ${env.UBUNTU_GNU_SRW_IMAGE_TAG}"
+        sh 'docker push "${UBUNTU_GNU_SRW_IMAGE_TAG}"'
       }
     }
 
-    stage('Build Ubuntu GNU SRW') {
-      steps {
-        echo "Building ${env.UBUNTU_GNU_SRW_IMAGE_TAG}"
-        sh 'docker build --tag "${UBUNTU_GNU_SRW_IMAGE_TAG}" --file "${WORKSPACE}/Docker/Dockerfile.${UBUNTU_GNU_SRW_IMAGE_NAME}" "${WORKSPACE}"'
+    
+    stage('RUN DATA SCRIPT') {
+        steps {
+          echo "Pulling input-data-20220414, Creating Docker image & Pushing it to DockerHub"
+          sh 'PASSWORD= ${DockerHub_PASSWORD} /path/to/your/aws-s3.sh'
```

Comment:
You don't need the `PASSWORD= ${DockerHub_PASSWORD}` part, you'll set an environment variable and import the credential from the Jenkins credential store. Check out the **Handling credentials** section in https://www.jenkins.io/doc/book/pipeline/jenkinsfile/. I can help you with this too.

Also, use `"${WORKSPACE}/aws-s3.sh"` for the path. Although, I'm not sure if the script should be checked in at the root of the repo, maybe in a `scripts` folder? @jkbk2004